src/List/ItemList.js

```diff
@@ -1,11 +1,16 @@
-import React, { useState, useEffect, Fragment } from "react";
+import React, { useEffect, Fragment, useState } from "react";
 import { NavLink } from "react-router-dom";
 import "firebase/firestore";
 import * as firebase from "../lib/firebase";
-import classes from "./List.module.css";
+import Modal from "../Modal/Modal";
+
+//this will be used to pass a promise around
+let resolve;
```

Comment:
I think there is simpler implementation of this where you avoid using `Promise` at all to handle the confirmation modal. I would say confirming/denying your action via that modal is a `synchronous` action.